src/xrpld/app/wasm/detail/HostFuncWrapper.cpp

```diff
@@ -186,6 +187,41 @@ getDataCurrency(IW const* runtime, wasm_val_vec_t const* params, int32_t& i)
     return Currency::fromVoid(slice->data());
 }
 
+template <class IW>
+static Expected<Asset, HostFunctionError>
+getDataAsset(IW const* runtime, wasm_val_vec_t const* params, int32_t& i)
+{
+    auto const slice = getDataSlice(runtime, params, i);
+    if (!slice)
+    {
+        return Unexpected(slice.error());
+    }
+
+    if (slice->size() == MPTID::bytes)
+    {
+        auto const mptid = MPTID::fromVoid(slice->data());
+        return Asset{mptid};
+    }
+
+    if (slice->size() == Currency::bytes)
+    {
+        auto const currency = Currency::fromVoid(slice->data());
+        auto const issue = Issue{currency, xrpAccount()};
+        if (!issue.native())
+            return Unexpected(HostFunctionError::INVALID_PARAMS);
+        return Asset{issue};
+    }
+
+    if (slice->size() == (AccountID::bytes + Currency::bytes))
+    {
+        return Asset{Issue(
```

Comment:
Should there be check   `if (issue.native()) return Unexpected(HostFunctionError::INVALID_PARAMS);` (opposite to previous one) ?

Reply:
[1f7df92](https://github.com/XRPLF/rippled/pull/5674/commits/1f7df924d8e3c3683fd3d8ba3e6ba5ae4828753d)